### ISSUE TITLE
1.10.2 Infinite resources

### ISSUE THREAD

**Issue by deni500** [lorddusk/HQM#233]
When you craft something in crafting table you can take back all material and you can craft it again and again ... again ...

**Comment by lorddusk** [comment]
How can you be sure it's HQM that's doing it?

**Comment by deni500** [comment]
Because when I removed it, Everything were fine and I am not only one who find that bug .. someone did video and post it on your project side

**Comment by lorddusk** [comment]
link?

**Comment by deni500** [comment]
https://www.youtube.com/watch?v=d4RFOZCOA4E

**Comment by mobehun** [comment]
I can confirm this too. With version HQM-New Heights-1.10.2-5.0.0beta4 and forge-12.18.1.2089
Here's the whole mod-list and the modpack too. Just in case...
https://github.com/mobehun/BraveTechnology/blob/master/AUTHORS
https://minecraft.curseforge.com/projects/brave-technology-new-world

I'll try a previous version as soon as I have time.
Edit:
I've checked it and it does happen with previous version too.

**Comment by Farronox** [comment]
Tested Sep 29/2016. crafting resources duplicate when crafted item is the target of a crafting task.

**Comment by sebbasel** [comment]
I have the same setting than Mobehun... and the same issue described in video by Deni500.
Is it related to this problem ? https://github.com/lorddusk/HQM/issues/204

In order to help developers, here my log :
http://pastebin.com/NHpDnvgA

**Comment by sebbasel** [comment]
I investigated the problem. It is caused by corrupted data in the 'data.json' file. There is a mismatch the number of items requested by the quest and the saved progress of it.

A comparison between the quest json file and the data json file :
http://pastebin.com/Y0VnVqhn

sample:

> {
>   "uuid": "1f231452-c6a9-4fb0-bbb5-f80c1a50a6b9",
>   "name": "ProjectE",
>   "description": "Having infinite amount of items is not an easy task. Before we start on that, let�s get infinite EMC, right?What do we need? Cobblestone!",
>   "x": 23,
>   "y": 36,
>   "icon": {
>     "id": "storagedrawers:compDrawers"
>   },
>   "optionlinks": [
>     "9ea50513-8c6f-4bac-bf4f-8d4ddf53e21f"
>   ],
>   "tasks": [
>     {
>       "type": "CRAFT",
>       "longDescription": "One way to make cobblestone easily is to use EU2 to do it for you.",
>       "items": [
>         {
>           "item": {
>             "id": "extrautils2:Ingredients",
>             "damage": 8
>           }
>         },
>         {
>           "item": {
>             "id": "extrautils2:Ingredients",
>             "damage": 6
>           },
>           "required": 20
>         },
>         {
>           "item": {
>             "id": "extrautils2:Grocket"
>           }
>         }
>       ]
>     },
>     {
>       "type": "CRAFT",
>       "longDescription": "Simply storing it would be wasteful. Compress it down all the way to Octuple Compressed.",
>       "items": [
>         {
>           "item": {
>             "id": "storagedrawers:compDrawers"
>           },
>           "required": 3
>         }
>       ]
>     },
>     {
>       "type": "DETECT",
>       "longDescription": "Now if only I could have a system that makes it all.",
>       "items": [
>         {
>           "item": {
>             "id": "minecraft:cobblestone"
>           }
>         },
>         {
>           "item": {
>             "id": "minecraft:lava_bucket"
>           }
>         },
>         {
>           "item": {
>             "id": "minecraft:water_bucket"
>           }
>         }
>       ]
>     }
>   ],
>   "commandrewards": []
> }
>  
> --- vs --- vs --- vs ---
>  
> {
>   "questId": "1f231452-c6a9-4fb0-bbb5-f80c1a50a6b9",
>   "questData": {
>     "players": 1,
>     "rewards": [
>       false
>     ],
>     "completed": false,
>     "claimed": false,
>     "available": true,
>     "time": 0,
>     "tasksSize": 3,
>     "tasks": [
>       {
>         "type": "ITEMS",
>         "completed": false,
>         "count": 3,
>         "progress": [
>           0,
>           0,
>           0
>         ]
>       }
>     ]
>   }
> }

**Comment by DaNeo61** [comment]
same here getting following error message on log:

> [17:59:49] [Server thread/ERROR] [FML/]: Exception caught during firing event net.minecraftforge.fml.common.gameevent.PlayerEvent$ItemCraftedEvent@32d5a30c:
> java.lang.ArrayIndexOutOfBoundsException: 1
>     at hardcorequesting.quests.task.QuestTaskItems.increaseItems(QuestTaskItems.java:311) ~[QuestTaskItems.class:?]
>     at hardcorequesting.quests.task.QuestTaskItemsCrafting.create(QuestTaskItemsCrafting.java:45) ~[QuestTaskItemsCrafting.class:?]
>     at hardcorequesting.quests.task.QuestTaskItemsCrafting.onCrafting(QuestTaskItemsCrafting.java:33) ~[QuestTaskItemsCrafting.class:?]
>     at hardcorequesting.event.EventHandler.onEvent(EventHandler.java:79) ~[EventHandler.class:?]
>     at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_705_EventHandler_onEvent_ItemCraftedEvent.invoke(.dynamic) ~[?:?]
>     at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90) ~[ASMEventHandler.class:?]
>     at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:185) [EventBus.class:?]
>     at net.minecraftforge.fml.common.FMLCommonHandler.firePlayerCraftingEvent(FMLCommonHandler.java:585) [FMLCommonHandler.class:?]
>     at net.minecraft.inventory.SlotCrafting.func_82870_a(SlotCrafting.java:111) [acb.class:?]
>     at net.minecraft.inventory.Container.func_184996_a(Container.java:324) [abd.class:?]
>     at slimeknights.mantle.inventory.ContainerMultiModule.func_184996_a(ContainerMultiModule.java:120) [ContainerMultiModule.class:?]
>     at net.minecraft.network.NetHandlerPlayServer.func_147351_a(NetHandlerPlayServer.java:1112) [me.class:?]
>     at net.minecraft.network.play.client.CPacketClickWindow.func_148833_a(SourceFile:33) [ir.class:?]
>     at net.minecraft.network.play.client.CPacketClickWindow.func_148833_a(SourceFile:10) [ir.class:?]
>     at net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13) [fl$1.class:?]
>     at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_25]
>     at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_25]
>     at net.minecraft.util.Util.func_181617_a(SourceFile:45) [h.class:?]
>     at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:668) [MinecraftServer.class:?]
>     at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:613) [MinecraftServer.class:?]
>     at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:149) [bzl.class:?]
>     at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:471) [MinecraftServer.class:?]
>     at java.lang.Thread.run(Thread.java:745) [?:1.8.0_25]

**Comment by royalcw** [comment]
I was getting a conflict between NetherPortalFix and this.  When I removed NetherPortalFix mod then the duplication went away.  The individuals who are experiencing the duping above...do you have NetherPortalFix mod in your game?

**Comment by NicosaurusRex99** [comment]
I got it as well http://pastebin.com/hcnzXjWc

**Comment by canitzp** [comment]
I investigated it and inside my dev env there isn't such a bug as described. Please can anyone can test this with only hqm installed?

**Comment by whatthedrunk** [comment]
It doesn't happen on a fresh world with only HQM installed but as soon as you start making the quests the duping starts happening and the ticking player crash I posted in the other issue. [Here is my HQM config if you want to see for yourself. ](https://drive.google.com/open?id=0ByQjJOciCO7lRGpvSzZmN0NHZG8)

**Comment by royalcw** [comment]
This issue is still existing in 1.12.2.  It is definitely HQM causing it with the same error code shown above.

**Comment by noobanidus** [comment]
@royalcw Is this still happening for you with the latest version released?

**Comment by noobanidus** [comment]
HQM is no longer actively developed for 1.11. While there was a comment over a year ago indicating this issue still occurred, please open a new issue if that is indeed the case.
